router/core/graphql_prehandler.go

```diff
@@ -315,11 +317,29 @@ func (h *PreHandler) Handler(next http.Handler) http.Handler {
 			body:             body,
 		})
 		if err != nil {
-			requestContext.error = err
+			outputErr := err
+			var httpErr ApolloCompatibleHttpError
+			var reportErr ReportError
+			if h.executor.Resolver.Options().ApolloCompatibilityReplaceInvalidVariableError && errors.As(err, &httpErr) {
+				outputErr = &operationreport.ApolloCompatibilityError{
+					ExtensionCode: httpErr.ExtensionCode(),
+					Message:       httpErr.Error(),
+				}
+			} else if h.executor.Resolver.Options().ApolloCompatibilityReplaceUndefinedOperationFieldError && errors.As(err, &reportErr) {
+				report := reportErr.Report()
+				for _, externalErr := range report.ExternalErrors {
+					if externalErr.ApolloCompatibilityError != nil {
+						outputErr = externalErr.ApolloCompatibilityError
+						break
+					}
+				}
+			}
+
+			requestContext.error = outputErr
```

Comment:
`requestContext.error` should be the original error from `handleOperation`.

Reply:
The reason all the logic is here is because I thought the error should match what is actually provided to the user. So it's fine for the errors to be completely different?